### PR TITLE
sidekiq added to the final project

### PR DESCRIPTION
**Pull request by archisss**
Added the sidekiq gem and the avatar image to the project

### CODE REVIEW

**Review by kikiondo — APPROVED** [review]
Hasta donde llegan mis conocimientos lo veo bien, no se si hay buenas practicas que me estoy saltando pero el funconamiento es correcto (y)

**Review by Mildred14 — APPROVED** [review]

File: app/controllers/todos_controller.rb [hunk 1]
```diff
@@ -5,7 +5,9 @@ class TodosController < ApplicationController
   # GET /todos
   # GET /todos.json
   def index
-    @todos = Todo.all
+    @todos = Todo.where(user_id: current_user.id)
+    #@todos = Todo.all
+    #@todos = Todo.find(:all, :conditions => { :user_id => current_user.id })
```
Comment (fernandoamz) [inline app/controllers/todos_controller.rb]: this comments are necessary ?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -12,6 +12,10 @@
 <body>
 <ul>
     <% if user_signed_in? %>
+    <% if current_user.avatar.attached? %>
```
Comment (fernandoamz) [inline app/views/layouts/application.html.erb]: this could be in a helper

File: app/controllers/todos_controller.rb [hunk 3]
```diff
@@ -42,6 +44,8 @@ def create
         format.html { render :new }
         format.json { render json: @todo.errors, status: :unprocessable_entity }
       end
+      #se puede enviar un email despues de crear un TODO
```
Comment (fernandoamz) [inline app/controllers/todos_controller.rb]: same issue here

File: app/workers/first_worker.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class FirstWorker
+
+  include Sidekiq::Worker
+
+  def perform(*args)
+    # Do something
```
Comment (fernandoamz) [inline app/workers/first_worker.rb]: this comment is necessary ?

File: config/application.rb [hunk 5]
```diff
@@ -16,5 +16,10 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+
+    #comando para poder utilizar el mailer
+    #config.action_mailer.default_url_options = { host: 'example.com'}
+
+    config.cache_store = :redis_store, "redis://localhost:6379/0/cache", { expires_in: 90.minutes }
   end
 end
```
Comment (fernandoamz) [inline config/application.rb:25]: add a blank line in finish

File: app/controllers/todos_controller.rb [hunk 6]
```diff
@@ -5,7 +5,9 @@ class TodosController < ApplicationController
   # GET /todos
   # GET /todos.json
   def index
-    @todos = Todo.all
+    @todos = Todo.where(user_id: current_user.id)
```
Comment (fernandoamz) [inline app/controllers/todos_controller.rb:8]: you can do this with a scope

**Review by fernandoamz — DISMISSED** [review]
there is things that could be better

File: test/mailers/welcome_mailer_test.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+require 'test_helper'
```
[the remark this answers is not in the document]
Comment (archisss) [reply test/mailers/welcome_mailer_test.rb:2]: i all ready change it

File: app/controllers/todos_controller.rb [hunk 8]
```diff
@@ -42,6 +41,7 @@ def create
         format.html { render :new }
         format.json { render json: @todo.errors, status: :unprocessable_entity }
       end
+      # TodoMailer.weekly_report(@todo).delivery_now
```
Comment (pacocd) [inline app/controllers/todos_controller.rb:44]: is this comment required?

File: app/views/todos/show.html.erb [hunk 9]
```diff
@@ -1,14 +1,15 @@
 <p id="notice"><%= notice %></p>
-
+<p></p>
```
Comment (pacocd) [inline app/views/todos/show.html.erb]: is this empty ```p``` label required?

**Comment by pacocd** [comment]
@archisss I see that you've been working in different features in this PR. Ideally, we should separate each feature in a specific PR.

For now we can merge this branch, but we have to take care to not mix features in a single PR or it will be rejected.

File: app/controllers/todos_controller.rb (hunk 8 above)
Comment (archisss) [reply app/controllers/todos_controller.rb:44]: sorry i'm working in this feature thats why the comment

File: app/views/todos/show.html.erb (hunk 9 above)
Comment (archisss) [reply app/views/todos/show.html.erb]: fixed

**Review by kikiondo — APPROVED** [review]
The last problem was complete successfully. I don't know if you have some bad practices in your code but for me is nice n.n jejeje

**Review by archisss — COMMENTED** [review]
ok

**Review by 8geonirt — APPROVED** [review]
🤓 👍

**Review by diegosaul402 — APPROVED** [review]
Like @pacocd said, you have too many features in just one pull request, this a bad practice that you shouldn't repeat, we'll allow you to merge this time.
Also, you have to add more descriptive commit messages, that's another reason to block PR merging.

Besides that, I have to congratulate you, you have been working very well, keep the hard work!
If you have any doubts ask any of us.
:shipit:

**Pull request merged**